nano/secure/ledger.hpp

```diff
@@ -51,7 +51,8 @@ class ledger final
 	nano::uint128_t account_balance (store::transaction const &, nano::account const &, bool = false);
 	nano::uint128_t account_receivable (store::transaction const &, nano::account const &, bool = false);
 	nano::uint128_t weight (nano::account const &);
-	std::shared_ptr<nano::block> successor (store::transaction const &, nano::qualified_root const &);
+	std::optional<nano::block_hash> successor (store::transaction const &, nano::qualified_root const &) const noexcept;
```

Comment:
There's an ambiguity there that appeared when epoch blocks were created. Roots are only used for block PoW and vote_req hinting. The successor function gives precedence to the block hash rather than the account number. The net result is that vote_req root hints don't work for epoch-open blocks.